universal-login-relayer/test/integration/ethereum/RelayerRequestSignatureValidator.test.ts

```diff
@@ -1,49 +1,53 @@
 import {expect} from 'chai';
-import {Wallet} from 'ethers';
+import {Wallet, utils} from 'ethers';
 import {createMockProvider, getWallets} from 'ethereum-waffle';
-import {TEST_GAS_PRICE, createKeyPair, signRelayerRequest, TEST_PRIVATE_KEY, recoverFromRelayerRequest, ETHER_NATIVE_TOKEN, EMPTY_DEVICE_INFO} from '@universal-login/commons';
+import {signRelayerRequest, TEST_PRIVATE_KEY, recoverFromRelayerRequest} from '@universal-login/commons';
 import RelayerRequestSignatureValidator from '../../../src/integration/ethereum/validators/RelayerRequestSignatureValidator';
-import setupWalletService, {createFutureWallet} from '../../testhelpers/setupWalletService';
 import createGnosisSafeContract from '../../testhelpers/createGnosisSafeContract';
 import {signStringMessage, calculateGnosisStringHash} from '@universal-login/contracts';
 import {setupWalletContractService} from '../../testhelpers/setupWalletContractService';
+import createWalletContract from '../../testhelpers/createWalletContract';
 
 describe('INT: RelayerRequestSignatureValidator', () => {
   let relayerRequestSignatureValidator: RelayerRequestSignatureValidator;
   let wallet: Wallet;
   let provider;
-  let contractAddress: string;
-  const keyPair = createKeyPair();
-  const ensName = 'jarek.mylogin.eth';
 
   before(async () => {
     provider = createMockProvider();
     [wallet] = await getWallets(provider);
     const walletContractService = setupWalletContractService(provider);
-    const {walletService, factoryContract, ensService} = await setupWalletService(wallet);
     relayerRequestSignatureValidator = new RelayerRequestSignatureValidator(walletContractService);
-    const {futureContractAddress, signature} = await createFutureWallet(keyPair, ensName, factoryContract, wallet, ensService);
-    await walletService.deploy({publicKey: keyPair.publicKey, ensName, gasPrice: TEST_GAS_PRICE, signature, gasToken: ETHER_NATIVE_TOKEN.address}, EMPTY_DEVICE_INFO);
-    contractAddress = futureContractAddress;
   });
 
   it('do not throw exception', async () => {
-    await expect(relayerRequestSignatureValidator.ensureValidRelayerRequestSignature(signRelayerRequest({contractAddress}, keyPair.privateKey))).to.be.fulfilled;
+    const {proxy} = await createWalletContract(wallet);
+    await expect(relayerRequestSignatureValidator.ensureValidRelayerRequestSignature(signRelayerRequest({contractAddress: proxy.address}, wallet.privateKey))).to.be.fulfilled;
   });
 
   it('throw exception', async () => {
-    const relayerRequest = signRelayerRequest({contractAddress}, TEST_PRIVATE_KEY);
+    const {proxy} = await createWalletContract(wallet);
```

Comment:
move to beforeEach?

Reply:
I am not sure if it is necessary. Have you seen that there are used two different functions `createWalletContract` and `createGnosisSafeContract` to deploy proxy?

Reply:
Maybe i losed it